### PR TITLE
[12.0][ADD] Report picking summary

### PR DESCRIPTION
**Pull request by quentinDupont**
This module adds the possibility to select several Pickings and print a Report summary with two sheets :
* Products Summary (where the total sum is calculated for each product)
* Detail Part (each Picking is shown)

Wizard to choose options : 
wizard

And print reportn this one with prices

pdf_report

This one without prices
pdf_report_wo_prices

Migration and improvment of @grap custom module from @legalsylvain : https://github.com/grap/grap-odoo-custom/tree/8.0/grap_qweb_report

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
Used in production since several monthes.
works like a charm.

code review & test.

Note @quentinDupont : you can add you as the maintainer of the application in the manifest adding the key : ``"maintainers": ["quentinDupont"],``
it will ping you if people want to change code, and you can also apply patches.

File: stock_picking_report_summary/models/picking_summary_wizard.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                    old_qty = product_lines[move.product_id.id]['qty']
+                    product_lines[move.product_id.id] = {
+                        'name': move.product_id.name,
+                        'categ': move.product_id.categ_id.name.capitalize(),
+                        'qty': old_qty + move.product_qty
+                    }
+        # Arranged in alphabetical order for category then product name
+        product_lines_sorted = sorted(
+            product_lines.items(), key=lambda x: (x[1]['categ'], x[1]['name']))
+        # product_lines_sorted = sorted(
```
Comment (rousseldenis) [inline stock_picking_report_summary/models/picking_summary_wizard.py]: Could you remove unused comments ?

File: stock_picking_report_summary/models/picking_summary_wizard.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        #     key=lambda x: )
+        for product_id, name_qty in product_lines_sorted:
+            res.append((0, 0, {
+                'product_id': product_id,
+                'quantity_total': name_qty['qty'],
+            }))
+        return res
+
+    # Columns Section
+    print_summary = fields.Boolean(
```
Comment (rousseldenis) [inline stock_picking_report_summary/models/picking_summary_wizard.py]: Could you put fields on top ?

File: stock_picking_report_summary/models/picking_summary_wizard.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        compute='_compute_standard_price_total',
+        digits=dp.get_precision('Product Unit of Measure'))
+
+    picking_line_ids = fields.One2many(
+        comodel_name='picking.summary.wizard.picking',
+        inverse_name='wizard_id', default=_default_picking_line_ids)
+
+    picking_line_qty = fields.Integer(
+        string='Number of Selected Picking', readonly=True,
+        default=_default_picking_line_qty)
```
Comment (rousseldenis) [inline stock_picking_report_summary/models/picking_summary_wizard.py]: You can change to lambda

File: stock_picking_report_summary/models/picking_summary_wizard.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        string='Print Detail', default=True)
+
+    print_unity_in_list = fields.Boolean(
+        string='Print Unit in Pickings List', default=True)
+
+    product_line_ids = fields.One2many(
+        comodel_name='picking.summary.wizard.product',
+        inverse_name='wizard_id', default=_default_product_line_ids)
+
+    standard_price_total = fields.Float(
```
Comment (rousseldenis) [inline stock_picking_report_summary/models/picking_summary_wizard.py]: I don't think this is common to add price in picking report (e.g.: if it's transmitted to customer as price can vary)

File: stock_picking_report_summary/reports/report_print_picking_summary_template.xml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                                        <span t-field="product_line.quantity_total"/>
+                                    </td>
+                                    <td class="align-center">
+                                        <span t-field="product_line.product_id.uom_id.name"/>
+                                    </td>
+                                    <td class="align-center">
+                                        <span t-esc="round(product_line.product_id.standard_price,2)"/>
+                                    </td>
+                                    <td class="align-center">
+                                        <span t-esc="round(product_line.standard_price_total,2)"/>
```
Comment (rousseldenis) [inline stock_picking_report_summary/reports/report_print_picking_summary_template.xml]: Why 2 ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
@quentinDupont Thanks for your contribution.

Code review.

**Comment by rousseldenis** [comment]
@quentinDupont

File: stock_picking_report_summary/models/picking_summary_wizard.py (hunk 4 above)
Comment (quentinDupont) [reply stock_picking_report_summary/models/picking_summary_wizard.py]: You're right !
It's now optional on wizard, by default set to False

File: stock_picking_report_summary/reports/report_print_picking_summary_template.xml (hunk 5 above)
Comment (quentinDupont) [reply stock_picking_report_summary/reports/report_print_picking_summary_template.xml]: I remove it and it's handled in fields precision

**Comment by quentinDupont** [comment]
> @quentinDupont

Thanks for your nice review ! I think i covered all your remarks :)

**Comment by quentinDupont** [comment]
@rousseldenis I think my last commit covers all your remarks if you can take a look ;)

**Review by rousseldenis — APPROVED** [review]
Seems very good.

**Review by legalsylvain — APPROVED** [review]
Thanks !

**Comment by rousseldenis** [comment]
@quentinDupont Could you modify your last commit message with something like "[IMP] stock_report_picking_summary: optional print prices" ?

**Comment by quentinDupont** [comment]
> @quentinDupont Could you modify your last commit message with something like "[IMP] stock_report_picking_summary: optional print prices" ?

Yep, here it is

**Comment by pedrobaeza** [comment]
Isn't that part the same as https://github.com/OCA/stock-logistics-reporting/tree/12.0/stock_picking_report_valued ?

**Comment by quentinDupont** [comment]
> Isn't that part the same as https://github.com/OCA/stock-logistics-reporting/tree/12.0/stock_picking_report_valued ?

Hello,
stock_picking_report_valued provides Delivery slip report, kind of a official document, but this new module stock_picking_report_summary is a quick visual summary of what you have to prepare, product by product. then picking by picking. 
In the food industry, it is useful for example to prepare orders, to know quickly the quantity of salads ordered by my x customers, then to distribute these salads in such or such basket (I agree this last part is close to the Delivery slip module).

**Pull request merged**